src/preferences.py

```diff
@@ -32,8 +32,8 @@ def check_config(self, config):
         with open("config.json", 'r') as f:
             template = json.load(f)
         if set(config.keys()) != set(template.keys()):
-            config = utilities.remove_unused_config_keys(config, template)
-            config = utilities.add_new_config_keys(config, template)
+            config = utilities.remove_unused_config_keys(self, config, template)
+            config = utilities.add_new_config_keys(self, config, template)
         return config
```

Comment:
Ah, now I see the issue. Neither function actually uses `self` though, so perhaps it's better to remove self from the input arguments in `utilities.py` instead.